File: python/135-single-number.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def singleNumber(self, nums):
        res = nums[0]
        for i, el in enumerate(nums):
            if i > 0:
                res = res ^ el
            else:
                res = el
        return res

File: javascript/src/135-single-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    var res = nums[0];
    for (var i=nums.length; i > 1; i--) {
        res = res ^ nums[i-1];
    }
    return res;
};

module.exports = singleNumber;